src/mscorlib/shared/System/Convert.cs

```diff
@@ -2656,7 +2656,48 @@ public static bool TryFromBase64String(string s, Span<byte> bytes, out int bytes
             return TryFromBase64Chars(s.AsSpan(), bytes, out bytesWritten);
         }
 
-        public static unsafe bool TryFromBase64Chars(ReadOnlySpan<char> chars, Span<byte> bytes, out int bytesWritten)
+        public static bool TryFromBase64Chars(ReadOnlySpan<char> chars, Span<byte> bytes, out int bytesWritten)
+        {
+            // First, attempt the decode using a faster algorithm that doesn't tolerate whitespace. 
+            if (Base64.TryDecodeFromUtf16(chars, bytes, out int consumed, out bytesWritten))
+            {
+                Debug.Assert(consumed == chars.Length);
+                return true;
+            }
+
+            Debug.Assert((consumed % 4) == 0);
+            if ((bytesWritten % 3) == 0)
+            {
+                // If we got here, the last valid 4-character chunk read had no padding characters ('='). It is possible that 
+                // the remainder of the string contains a valid Base64 encoding with whitepace interspersed. Continue decoding the rest of
+                // the string using the slower algorithm that ignores whitespace.
+                if (!TryFromBase64CharsSlow(chars.Slice(consumed), bytes.Slice(bytesWritten), out int moreBytesWritten))
```

Comment:
Should we rather skip the whitespace here, and go back to the fast TryDecodeFromUtf16 ?

Reply:
Hmm... I like that idea.

Reply:
Wait, it's more complicated than that - if the whitespace occurs in the middle of a 4-byte chunk, the fast parser stops at the end of the prior 4-byte chunk - i.e. somebody still has to process up to 3 possible Base64 characters before the whitespace starts.

Reply:
We could still do this if "consumed % 4 == 0" - assuming that's the typical case where a Base64 string is formatted into multilines (i.e. each line contains N*4 base64 chars.)

Reply:
And for the other case, you may be able to copy over the next non-whitespace chars to temp buffer.

I think it would be really nice to get rid of `FromBase64_Decode` as part of this change because of most of this is duplicate of the code you are adding.